ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -798,77 +724,172 @@ ttnn::operations::matmul::MatmulProgramConfig determine_matmul_op_config_from_co
+            // be conservative with L1 memory usage.
+            // act_block_w_div == 1 is currently the default value.
+            conv_config.act_block_w_div = tt::div_up(in_channels, width_sharded_num_cores * constants::TILE_WIDTH);
+        }
+
+        conv_op_l1_usage l1_usage = calculate_L1_usage(
+            arch,
+            conv_config.shard_layout.value(),
+            input_dtype,
+            conv_config.weights_dtype,
+            conv_config.dtype,
+            compute_config,
+            opt_conv_op_block_config,
+            opt_conv_op_parallel_config,
+            in_channels,
+            out_channels,
+            weights_shape,
+            batch_size,
+            output_width,
+            groups,
+            kernel_size,
+            conv_config,
+            conv_out_memory_config,
+            enable_bias,
+            use_non_tile_height);
+        log_debug(
+            tt::LogOp,
+            "L1 usage for {}: {}, {}",
+            conv_config.shard_layout,
+            l1_usage.tensor_allocation_size,
+            l1_usage.CB_allocation_size);
+        return core_count_and_size{
+            .core_count = input_parallel_config.grid.num_cores(),
+            .size = l1_usage.CB_allocation_size + l1_usage.tensor_allocation_size,
+            .conv_config = conv_config};
+    };
+
+    core_count_and_size height = get_l1_usage_for_sharding(TensorMemoryLayout::HEIGHT_SHARDED, conv_config);
+
+    // 1d convs support only height sharding
+    const bool is_conv1d = weights_width == 1 && input_width == 1;
+    if (is_conv1d) {
+        return height.conv_config;
     }
 
-    if (conv_config.act_block_w_div == 1 && conv_config.shard_layout == TensorMemoryLayout::WIDTH_SHARDED) {
-        uint32_t width_sharded_num_cores = determine_parallel_config(
-                                               TensorMemoryLayout::WIDTH_SHARDED,
-                                               batch_size,
-                                               in_channels,
-                                               output_height,
-                                               output_width,
-                                               out_channels,
-                                               compute_grid_size,
-                                               shard_orientation,
-                                               !is_mm_conv)
-                                               .grid.num_cores();
-        // Set act_block_w_div to max value to
-        // be conservative with L1 memory usage.
-        // act_block_w_div == 1 is currently the default value.
-        conv_config.act_block_w_div = tt::div_up(in_channels, width_sharded_num_cores * constants::TILE_WIDTH);
+    core_count_and_size block = get_l1_usage_for_sharding(TensorMemoryLayout::BLOCK_SHARDED, conv_config);
+    core_count_and_size weight = get_l1_usage_for_sharding(TensorMemoryLayout::WIDTH_SHARDED, conv_config);
```

Comment:
Should probably be width (and const)

```suggestion
    const core_count_and_size width = get_l1_usage_for_sharding(TensorMemoryLayout::WIDTH_SHARDED, conv_config);
```

Reply:
thx